### ISSUE TITLE
Error "partitioning failed; increase redMemory." during assembly

### ISSUE THREAD

**Issue by Milokita** [marbl/canu#2179]
When tried to assemble a low coverage covid genome, the following error was promoted:

```
Detected 24 CPUs and 126 gigabytes of memory on the local machine.
Local machine mode enabled; grid support not detected or not allowed.

BEGIN ASSEMBLY

Loading read lengths.
Loading number of overlaps per read.
Configure RED for 16gb memory.
                  Batches of at most (unlimited) reads.
                                      500000000 bases.
                   Expecting evidence of at most 536870912 bases per iteration.

           Total                                               Reads                 Olaps Evidence
    Job   Memory      Read Range         Reads        Bases   Memory        Olaps   Memory   Memory  (Memory in MB)
  ---- -------- ------------------- --------- ------------ -------- ------------ -------- --------
      1  3088.35         1-85               67       133533    16.30         4260     0.05  1024.00
  ---- -------- ------------------- --------- ------------ -------- ------------ -------- --------
                                                     133533                  4260

ABORT:
ABORT: canu 2.2
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:
ABORT:   partitioning failed; increase redMemory.
ABORT:
```

Command used: 
`canu -p asm.covid -d asm.covid genomeSize=5k correctedErrorRate=0.105 -nanopore mapped/drna.covid19.mapped.fastq`

**Comment by skoren** [comment]
Have you tried increasing redMemory as suggested, try adding redMemory=32 to your options and re-start.

**Comment by Milokita** [comment]
After I increased the redMemory as you suggested (because I did not know the config)
`canu -p asm.covid -d asm.covid genomeSize=5k  redMemory=126 -nanopore mapped/drna.covid19.mapped.fastq`

the same error is returned:

```
-- CONFIGURE CANU
--
-- Detected Java(TM) Runtime Environment '11.0.16' (from 'java') without -d64 support.
-- Detected gnuplot version '5.4 patchlevel 1   ' (from 'gnuplot') and image format 'png'.
--
-- Detected 24 CPUs and 126 gigabytes of memory on the local machine.
--
-- Local machine mode enabled; grid support not detected or not allowed.
--
--                                (tag)Concurrency
--                         (tag)Threads          |
--                (tag)Memory         |          |
--        (tag)             |         |          |       total usage      algorithm
--        -------  ----------  --------   --------  --------------------  -----------------------------
-- Local: meryl     12.000 GB    4 CPUs x   6 jobs    72.000 GB  24 CPUs  (k-mer counting)
-- Local: hap        8.000 GB    4 CPUs x   6 jobs    48.000 GB  24 CPUs  (read-to-haplotype assignment)
-- Local: cormhap    6.000 GB   12 CPUs x   2 jobs    12.000 GB  24 CPUs  (overlap detection with mhap)
-- Local: obtovl     4.000 GB    8 CPUs x   3 jobs    12.000 GB  24 CPUs  (overlap detection)
-- Local: utgovl     4.000 GB    8 CPUs x   3 jobs    12.000 GB  24 CPUs  (overlap detection)
-- Local: cor        -.--- GB    4 CPUs x   - jobs     -.--- GB   - CPUs  (read correction)
-- Local: ovb        4.000 GB    1 CPU  x  24 jobs    96.000 GB  24 CPUs  (overlap store bucketizer)
-- Local: ovs        8.000 GB    1 CPU  x  15 jobs   120.000 GB  15 CPUs  (overlap store sorting)
-- Local: red      126.000 GB    4 CPUs x   1 job    126.000 GB   4 CPUs  (read error detection)
-- Local: oea        8.000 GB    1 CPU  x  15 jobs   120.000 GB  15 CPUs  (overlap error adjustment)
-- Local: bat       16.000 GB    4 CPUs x   1 job     16.000 GB   4 CPUs  (contig construction with bogart)
-- Local: cns        -.--- GB    4 CPUs x   - jobs     -.--- GB   - CPUs  (consensus)
--
-- Found Nanopore reads in 'asm.covid.seqStore':
--   Libraries:
--     Nanopore:              1
--   Reads:
--     Raw:                   172073
--     Corrected:             137269
--     Corrected and Trimmed: 132382
--
--
-- Generating assembly 'asm.covid' in '/home/zx/gluster/nanopore.rna/czg.covid/drna.covid.2nd/covid19_ct13_restart1/20220802_2232_MN24113_FAN13542_14f78b6e/asm.covid':
--   genomeSize:
--     5000
--
--   Overlap Generation Limits:
--     corOvlErrorRate 0.3200 ( 32.00%)
--     obtOvlErrorRate 0.1200 ( 12.00%)
--     utgOvlErrorRate 0.1200 ( 12.00%)
--
--   Overlap Processing Limits:
--     corErrorRate    0.3000 ( 30.00%)
--     obtErrorRate    0.1200 ( 12.00%)
--     utgErrorRate    0.1200 ( 12.00%)
--     cnsErrorRate    0.2000 ( 20.00%)
--
--   Stages to run:
--     assemble corrected and trimmed reads.
--
--
-- Correction skipped; not enabled.
--
-- Trimming skipped; not enabled.
--
-- BEGIN ASSEMBLY
--
-- Loading read lengths.
-- Loading number of overlaps per read.
--
-- Configure RED for 126gb memory.
--                   Batches of at most (unlimited) reads.
--                                      500000000 bases.
--                   Expecting evidence of at most 536870912 bases per iteration.
--
--           Total                                               Reads                 Olaps Evidence
--    Job   Memory      Read Range         Reads        Bases   Memory        Olaps   Memory   Memory  (Memory in MB)
--   ---- -------- ------------------- --------- ------------ -------- ------------ -------- --------
--      1  3088.35         1-85               67       133533    16.30         4260     0.05  1024.00
--   ---- -------- ------------------- --------- ------------ -------- ------------ -------- --------
--                                                     133533                  4260

ABORT:
ABORT: canu 2.2
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:
ABORT:   partitioning failed; increase redMemory.
ABORT:

```

It seems very strange as there are only 250 covid reads, it should not require such large amount of memory

**Comment by skoren** [comment]
Ah, I think you're hitting #2035 which I thought was fixed in 2.2 but it postdates that fix. You can either update the code to use the latest if you can compile canu locally from GitHub. Alternatively, you need >100 reads after correction which you may be able to get by lowering the minimum read length/overlap.

**Comment by skoren** [comment]
Idle
